jina/drivers/__init__.py

```diff
@@ -477,6 +477,8 @@ def attach(self, executor: 'AnyExecutor', *args, **kwargs) -> None:
         if self._executor_name and isinstance(executor, CompoundExecutor):
             if self._executor_name in executor:
                 self._exec = executor[self._executor_name]
+            elif self._executor_name == executor.name:
```

Comment:
Does a `CompoundExecutor` have a `name` property? Why is this intended? Is it intended for a special subclass of `CompoundExecutor`?

Reply:
Removed